src/odb.rs

```diff
@@ -66,6 +68,70 @@ impl<'repo> Odb<'repo> {
             Ok(())
         }
     }
+
+    /// Read object from the database.
+    pub fn read(&self, oid: Oid) -> Result<OdbObject, Error> {
+        let mut out = ptr::null_mut();
+        unsafe {
+            try_call!(raw::git_odb_read(&mut out, self.raw, oid.raw()));
+            Ok(OdbObject::from_raw(out))
+        }
+    }
+
+    /// Checks if the object database has an object.
+    pub fn exists(&self, oid: Oid) -> Result<bool, Error> {
```

Comment:
Should this perhaps just return a `bool` instead of a `Result`? (not sure it can return an `Error`

Reply:
That's correct, I'll fix that.